Gender on user profile pull request

AlineTheAlien opened the pull request:
This pull request includes: 

- Implementation of the gender icon on a user profile. This takes the information of the user whose gender we are interested in from the server, and depending on the gender displays the appropriate icon next to the user's name on their user profile. The get and set gender mapper methods are also included. 

- Test cases for the get and set gender methods, that test whether the methods receive the correct response from the server. These tests are included in the UserMapperTests class.

- Refactoring of the user profile. The last user profile wasn't too aesthetically pleasing, therefore it was updated and modified to follow more conventional mobile application interface design. 

- User icon on the bottom toolbar of the menu bar can be tapped to access the user profile of the user who is currently using the app. This allows a user to view their own user profile while using the application.

AlineTheAlien committed:
MI-193 Implemented change of gender icon in user profile based on user who is currently using the application.

This will later be changed by adding a gender column in the User table, in order to allow us to  fetch the gender of any user whose profile we want to view.

AlineTheAlien committed:
MI-193 Added mapper methods to communicate with the server to obtain gender information, and to change a user's gender.

Also includes code that sets the display icon of gender on a user's profile depending on the gender of the user that is retrieved. Hasn't been tested yet as the server-side pull request isn't approved yet.

AlineTheAlien committed:
MI-192 Tested and fix bugs with the displaying gender on user profile feature. Since the string type cannot be initialized with the "new" keyword, the method to obtain gender in the UserMapper class cannot use the default sendGet method in the AbstractMapper class.

In addition, dded button on bottom toolbar for a user to access their own user profile from the user icon on the bottom menu. This opens their own user profile.

Also changed the sizes of the male and female icons as they were too large for the emulator resolution.

AlineTheAlien committed:
Resource designer -  committing automatically updated configuration file

AlineTheAlien committed:
MI-193 Test cases for the get and set gender mapper methods.

The get method tests for whether the user's gender is sucesfully obtained  from the server side. The set method tests for whether the user's gender can be succesfully modified.

Also includes implementation of setGender in the user mapper which was forgotten in the last commit.

AlineTheAlien committed:
MI-194 Updated the test suite for UserMapperTests class.

AlineTheAlien committed:
MI-200 Changes to the user profile user interface.

Includes resizing of the gender icons in order to fit the interface based on the resolution of the emulator.

Also includes changes in the placements of the elements on the UserProfile axml view. This is in order to fit longer names on the page, and view the gender right next to the user name, instead of below.

AlineTheAlien committed:
Automatically changed target framework version

AlineTheAlien committed:
Merge branch 'dev' of https://github.com/MI-390/mRides-app.git

AlineTheAlien committed:
Merging dev into MI192 branch.

Includes changes that were not previously merged.

AlineTheAlien committed:
MI-200 Changed appearance of user profile page by making the top half of the page have a background colour similar to the colour of the mRides logo.

Also includes small changes to the ratings and reviews appearance on the user profile.

AlineTheAlien committed:
Resolving conflicts by merging dev into this branch.

Conflicts were in the MainActivity class, where the login with facebook implementation used to be. However, in the most recent push to dev, a seperate class was created to handle to login requests to increase the cohesion of the classes. Therefore I moved the code I added to the new LoginRequest class.

NassimElSayed commented on mRides-app/UserProfileActivity.cs:
```diff
@@ -50,6 +50,26 @@ protected override void OnCreate(Bundle savedInstanceState)
             gsdText.Text = "$ " + userGSD.ToString() + " GSD";
 
+            ImageView genderIcon = FindViewById<ImageView>(Resource.Id.genderImage);
+            //userMapper.setGender(id, "male");
```
You should remove this comment

AlineTheAlien replied:
Totally missed it, thank you

NassimElSayed commented on mRides-app/MapActivity.cs:
```diff
@@ -440,6 +437,16 @@ protected override void OnResume()
+            userProfileButton.Click += delegate
+            {
+                Intent i = new Intent(this, typeof(UserProfileActivity));
+                i.PutExtra("id", User.currentUser.id.ToString()); //not sure if we store this userID
```
Don't forget to remove this one too

NassimElSayed commented on mRides-app/Mappers/UserMapper.cs:
```diff
@@ -199,6 +199,49 @@ public long setGSD(int id, long gsdAmount)
+        }
+
+        /**
+        * Change a user's gender
```
Change this to an XML comment for the doc

AlineTheAlien committed:
Removed commented out code, and added missing XML comments for this branch.

Changes also include changing the regular style comments to XML comments in the UserMapper class.
